### DISCUSSION TITLE
Path Simplification with Freehand Pencil Tool

### DISCUSSION THREAD

**Discussion by sethhoneycutt** [SVG-Edit/svgedit#636] [New feature request]
Would it be possible to add an option for path simplification with the freehand pencil tool to reduce the number of segments in the drawn path? When using the pencil tool it currently produces a large amount of data which makes the file size much larger. Simplifying the path makes for a smoother line and still follows closely to what was drawn while also reducing the amount of data for the path. Inkscape has a feature similar to this in the “Path > Simplify” option from the menu bar. If a feature like this could be added, perhaps a selection for the amount of simplification could be chosen down in the bottom toolbar along with the stroke width and opacity options. 

I found a demo of some JavaScript code for path simplification which shows what I am referring to:

[http://paperjs.org/examples/path-simplification/](http://paperjs.org/examples/path-simplification/)

**Comment by codedread** [comment]
Indeed - this would be awesome.  See the discussion 2 years ago: https://groups.google.com/forum/#!searchin/svg-edit/%22convert$20shapes$20to$20path%22/svg-edit/0a6UMEuTaBA/d3g5_VBbcesJ
